### PR TITLE
Introduce PHPStan

### PR DESCRIPTION
**Pull request by simPod**

### CODE REVIEW

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -0,0 +1,12 @@
+parameters:
+	ignoreErrors:
+		-
+			message: "#^Class JMS\\\\Serializer\\\\Handler\\\\iterable not found\\.$#"
```
Comment (goetas) [inline phpstan-baseline.neon:4]: hmm, this is weird... why do we get this error?

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+			count: 2
+			path: src/Handler/IteratorHandler.php
+
+		-
+			message: "#^Variable \\$previousEntityLoaderState might not be defined\\.$#"
```
Comment (goetas) [inline phpstan-baseline.neon:9]: was it possible to fix this?

**Review by goetas — COMMENTED** [review]
nice! i like it!

Can you please have a look at the two comments i've left?

File: phpstan-baseline.neon (hunk 1 above)
Comment (simPod) [reply phpstan-baseline.neon:4]: I don't know yet. Could not reproduce it on phpstan playground.

File: phpstan-baseline.neon (hunk 2 above)
Comment (simPod) [reply phpstan-baseline.neon]: check now

File: phpstan-baseline.neon (hunk 1 above)
Comment (goetas) [reply phpstan-baseline.neon:4]: maybe it needs just a `\`?

Comment (simPod) [reply phpstan-baseline.neon:4]: it's not an object so cannot be imported. It's like `string`.

https://phpstan.org/r/84508559-53e5-43da-b083-219d3b9af0a5

I was thinking that would investigate it later, that's why I added it to baseline for now.

File: phpstan.neon.dist [hunk 3]
```diff
@@ -0,0 +1,14 @@
+parameters:
+    level: 1
```
Comment (goetas) [inline phpstan.neon.dist:2]: hmm, level 1, @simPod do you plan in increasing the level with future pull requests?

Comment (simPod) [reply phpstan.neon.dist:2]: Yup

File: phpstan-baseline.neon (hunk 1 above)
Comment (Tobion) [reply phpstan-baseline.neon:4]: I found the problem, see https://github.com/schmittjoh/serializer/pull/1315

**Pull request merged**